code/lib/cli/src/automigrate/fixes/new-frameworks.ts

```diff
@@ -62,12 +62,15 @@ interface NewFrameworkRunOptions {
   };
 }
 
-export const getBuilder = (builder: string | { name: string }) => {
-  if (typeof builder === 'string') {
-    return builder.includes('vite') ? 'vite' : 'webpack5';
+const getBuilder = (builder: string | { name: string }, frameworkPackage: string) => {
+  const name = typeof builder === 'string' ? builder : builder?.name;
+  if (name?.includes('webpack')) {
+    return 'webpack5';
   }
-
-  return builder?.name.includes('vite') ? 'vite' : 'webpack5';
+  if (name?.includes('vite')) {
+    return 'vite';
+  }
+  return frameworkPackage.includes('svelte') || frameworkPackage.includes('vue3') ? 'vite' : 'webpack5';
```

Comment:
this should change from `vue3` to `vue` if https://github.com/storybookjs/storybook/pull/20970 is merged